models/classes/TaoLtiSession.php

```diff
@@ -89,12 +87,11 @@ public function getLaunchData()
      */
     public function getLtiLinkResource()
     {
-        if (is_null($this->ltiLink)) {
+        if (is_null($this->ltiLinkId)) {
             $service = $this->getServiceLocator()->get(LinkService::SERVICE_ID);
             $consumer = $this->getLaunchData()->getLtiConsumer();
-            $linkId = $service->getLinkId($consumer->getUri(), $this->getLaunchData()->getResourceLinkID());
-            $this->ltiLink = new core_kernel_classes_Resource($linkId);
+            $this->ltiLinkId = $service->getLinkId($consumer->getUri(), $this->getLaunchData()->getResourceLinkID());
         }
-        return $this->ltiLink;
+        return new core_kernel_classes_Resource($this->ltiLinkId);
```

Comment:
why do we need to have a new instance of the resource on every call?

Reply:
can we use `clone` instead of `new` here? it should be a bit faster

Reply:
This is about deserialization. It's better to serialize/deserialize an id thatn a resource object.

Reply:
@jbout Do we have existing serialized data that we've to take care about when changing the way we deserialize?

Reply:
I don't think it makes a big difference if we do not serialize `common_session_DefaultSession` in the 'proper' way